src-tauri/src/config.rs

```diff
@@ -0,0 +1,251 @@
+use serde::{Serialize, Deserialize};
+use home::{home_dir};
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use core::fmt::Display;
+
+const DEFAULT_CONFIG_FILENAME: &str = ".noters.yaml";
+
+pub struct Config {
+    filename: Box<Path>,
```

Comment:
Why work on head here?

Reply:
Because I lost the epic fight with the borrow checker ;-).
You cannot put a Path directory into a struct, since it's size isn't known at compile time. I had to choose between using a string (which is allocated as well) or using a boxed type. With the boxed type we don't loose the type.